docs/program/extend/modular-resources/_index.md

```diff
@@ -1,150 +1,528 @@
+For example:
 
-Name | Type | Default Value | Description
--------------- | ---- | ------------- | ---------------
-`name` | string | --        | Name of the module you are registering
-`executable_path` | string | --         | The filesystem path to the module executable on the robot part
+{{< tabs name="Template Shell Scripts as Module Executables" >}}
+{{% tab name="Go" %}}
 
-### Configure a component instance for a modular resource
+``` shell
+#!/bin/sh
+cd <path-to-your-module-directory>
 
-Once you have configured a module as part of your robot configuration, you can instantiate any number of instances of a modular resource made available by that module with the component or service configuration.
-All standard properties such as *attributes* and *depends_on* are supported for modular resources.
-To correctly reference a registered modular resource, you must configure the *namespace*, *type*, *name* and *model* properties.
+go build ./
+# Be sure to use `exec` so that termination signals reach the go process,
+# or handle forwarding termination signals manually
+exec ./<your-module-directory-name> $@
+```
 
-#### Required attributes - modular component
+{{% /tab %}}
+{{% tab name="Python" %}}
 
-Name | Type | Default Value | Description
--------------- | ---- | ------------- | ---------------
-`namespace` | string | --        | The namespace of the [API](#apis) (the first part of the [API](#apis) triplet)
-`type` | string | --         | The subtype of the [API](#apis) (the third part of the [API](#apis) triplet)
-`name` | string | --         | A unique name for this configured instance of the modular resource
-`model` | string | --         | The [full triplet](#models) of the modular resource
+``` shell
+#!/bin/sh
+cd <path-to-your-module-directory>
 
-## Configuration example - custom motor
+# Be sure to use `exec` so that termination signals reach the python process,
+# or handle forwarding termination signals manually
+exec python3 <your-module-directory-name>.main $@
+```
 
-The following example configuration illustrates how you can configure a robot that uses a custom motor implementation.
-This example motor implementation uses the built-in Viam motor API, but registers and uses a custom model __viam-contributor:motor:super-custom__.
+{{% /tab %}}
+{{< /tabs >}}
 
-``` json
+To make this file executable, run the following command in your terminal:
+
+``` shell
+sudo chmod +x <FILEPATH>/<FILENAME>
+```
+
+### Make sure `viam-server` can access your executable
+
+Ensure that the code defining your module is saved where the instance of `viam-server` behind your robot can read and execute it.
+
+For example, if you are running `viam-server` on an Raspberry Pi, you'll need to save the module on the Pi's filesystem.
+
+Obtain the path to the executable file on your computer's filesystem by running the following commands in your terminal:
+
+``` shell
+cd <path-to-your-module-directory>
+pwd
```

Comment:
This gives you the directory where your module is, not the full path (that goes in the config) which includes the directory AND the name of the module file.

Reply:
Ok addressed I think!

Reply:
No, I think there was a misunderstanding. 

What I originally meant is that this shell script is not doing what you say it is. The text says it is how you "obtain the path to the executable file", but it's not doing that. It is, in fact, only obtaining the part of the path; the directory the executable is in. You still must append the executable name. You can't just cd to the executable itself either (per this change), since it's not a directory.

You probably just want to instruct people to "get the full path to your executable" and let them know that if they see it in their terminal, they can use realpath. E.g. `realpath myModule` or `realpath moduleDirectory/myModule`

Reply:
Oh yes I did misunderstand this/wasn't paying attention to the rest of the instructions here-- thanks for the suggestion/fixed